### PR TITLE
Rickety quadjumper implementation and tests

### PR DESCRIPTION
**Pull request by sethhelman**

### CODE REVIEW

**Review by VJubert — DISMISSED** [review]

File: server/game/cards/02_SHD/units/RicketyQuadjumper.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+            internalName: 'rickety-quadjumper'
+        };
+    }
+
+    public override setupCardAbilities() {
+        this.addOnAttackAbility({
+            title: 'Reveal a card',
+            optional: true,
+            immediateEffect: AbilityHelper.immediateEffects.reveal((context) => ({ target: context.source.controller.getTopCardOfDeck() })),
+
```
Comment (AMMayberry1) [inline server/game/cards/02_SHD/units/RicketyQuadjumper.ts:17]: nit: shouldn't be an empty space here

File: server/game/cards/02_SHD/units/RicketyQuadjumper.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            ifYouDo: (context) => ({
```
Comment (AMMayberry1) [inline server/game/cards/02_SHD/units/RicketyQuadjumper.ts:18]: nit: we typically call this variable `ifYouDoContext` to distinguish from any other context that might be used deeper inside

File: test/server/cards/02_SHD/units/RicketyQuadjumper.spec.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+                        groundArena: ['wampa', 'battlefield-marine'],
+                        spaceArena: ['rickety-quadjumper'],
+                        deck: ['protector'],
+                    },
+                    player2: {
+                        groundArena: ['atst'],
+                    }
+                });
+
+
```
Comment (AMMayberry1) [inline test/server/cards/02_SHD/units/RicketyQuadjumper.spec.ts:18]: nit: should only be one newline, not two. handful of cases of this throughout the file

File: test/server/cards/02_SHD/units/RicketyQuadjumper.spec.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                // attack with rickety
+                context.player1.clickCard(context.ricketyQuadjumper);
+                context.player1.clickCard(context.p2Base);
+
+                // player1 should have prompt or pass
+                expect(context.player1).toHavePassAbilityPrompt('Reveal a card');
+                context.player1.clickPrompt('Reveal a card');
+
+                // top card is an upgrade, give exp to another unit
+                expect(context.protector).toBeInZone('deck');
```
Comment (AMMayberry1) [inline test/server/cards/02_SHD/units/RicketyQuadjumper.spec.ts:28]: Need to check that the reveal showed the expected card in the chat log. See usages of `getChatLogs()` in the C-3PO tests for an example

**Review by AMMayberry1 — APPROVED** [review]

**Pull request merged**